docs/examples/FormInputAddons.js

```diff
@@ -0,0 +1,71 @@
+const formInstance = (
+  <form>
+    <FormGroup>
+      <InputGroup>
+        <InputGroup.Addon>@</InputGroup.Addon>
+        <FormControl type="text" />
+      </InputGroup>
+    </FormGroup>
+    <FormGroup>
+      <InputGroup>
+        <FormControl type="text" />
+        <InputGroup.Addon>.00</InputGroup.Addon>
+      </InputGroup>
+    </FormGroup>
+    <FormGroup>
+      <InputGroup>
+        <InputGroup.Addon>$</InputGroup.Addon>
+        <FormControl type="text" />
+        <InputGroup.Addon>.00</InputGroup.Addon>
+      </InputGroup>
+    </FormGroup>
+
+    <FormGroup>
+      <InputGroup>
+        <FormControl type="text" />
+        <InputGroup.Addon>
+          <Glyphicon glyph="music" />
+        </InputGroup.Addon>
+      </InputGroup>
+    </FormGroup>
+
+    <FormGroup>
+      <InputGroup>
+        <InputGroup.Button>
+          <Button>Before</Button>
+        </InputGroup.Button>
+        <FormControl type="text" />
+      </InputGroup>
+    </FormGroup>
+    <FormGroup>
+      <InputGroup>
+        <FormControl type="text" />
+        {/* Set z-index here to work around some layout quirks. */}
+        <InputGroup.Button style={{ zIndex: 1000 }}>
```

Comment:
This is just https://github.com/react-bootstrap/react-bootstrap/issues/1283. I thought it was pretty gnarly to deal with properly. This is at least a workaround (that's now possible – previously it wasn't). Maybe I'll look at this a bit more later.